src/main/java/com/mathworks/ci/RunMatlabCommandBuilder.java

```diff
@@ -112,22 +116,40 @@ public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
     private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher,
             TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
         final String uniqueTmpFldrName = getUniqueNameForRunnerFile();
-        ProcStarter matlabLauncher;
+        final String uniqueCommandFile =
+                "command_" + getUniqueNameForRunnerFile().replaceAll("-", "_");
+
+        // Get unique temporary folder filepath
+        final FilePath uniqeTmpFolderPath =

```

Comment:
At the very least the creation of the temporary command file should be its own function so that we don't have all these implementation details here.

Reply:
Done. file creation is moved to separate method.